src/ivc/fold_relaxed_plonk_instance_chip.rs

```diff
@@ -531,43 +524,27 @@ where
 
         debug!("fold: Assigned & Generated challenge: {r:?}");
 
-        let gate = MainGate::new(config.clone());
-
-        let bn_chip = BigUintMulModChip::<C::Base>::new(
-            config
-                .into_smaller_size::<{ big_uint_mul_mod_chip::MAIN_GATE_T }>()
-                .unwrap(),
-            self.limb_width,
-            self.limbs_count,
-        );
+        let gate = MainGate::new(self.config.clone());
 
         let r_value = gate.le_bits_to_num(region, &r)?;
         let r = BigUintView {
-            as_bn_limbs: bn_chip.from_assigned_cell_to_limbs(region, &r_value)?,
+            as_bn_limbs: self.bn_chip.from_assigned_cell_to_limbs(region, &r_value)?,
             as_bits: r,
         };
 
-        let new_folded_W = Self::fold_W(region, config, &folded_W, &input_W_commitments, &r)?;
+        let new_folded_W = Self::fold_W(region, &self.config, &folded_W, &input_W_commitments, &r)?;
 
         debug!("fold: W folded: {new_folded_W:?}");
 
-        let new_folded_E = Self::fold_E(
-            region,
-            config,
-            &bn_chip,
-            folded_E,
-            &cross_terms_commits,
-            r.clone(),
-            &m_bn,
-        )?;
+        let new_folded_E = self.fold_E(region, folded_E, &cross_terms_commits, r.clone(), &m_bn)?;
         debug!("fold: E folded: {new_folded_W:?}");
 
         let new_folded_u = gate.add(region, &folded_u, &r_value)?;
         debug!("fold: u folded: {new_folded_u:?}");
 
         let [new_folded_X0, new_folded_X1] = Self::fold_instances(
```

Comment:
It's best to make the features as pure as possible. That is, creating a method is always worse than creating a [pure function](https://en.wikipedia.org/wiki/Pure_function)

That's why I used the following rule:
- If a function uses `bn_chip` + `config`, I make it a method so that I don't pass two parameters
- If the function uses only one thing from Self, then I make a pure function out of it

This is also more convenient in tests, when you don't have to create a chip to test a one fn